### PR TITLE
chore: upgrade to chokidar 4

### PR DESCRIPTION
**Pull request by benmccann**
Removes a dozen dependencies

### CODE REVIEW

**Comment by paulmillr** [comment]
Shouldn’t second arg be default = {} instead?

**Comment by benmccann** [comment]
Chokidar does that be internally in the watcher constructor, so it shouldn't need to be done elsewhere

**Comment by benmccann** [comment]
This PR is ready for review now

File: packages/svelte-check/src/index.ts [hunk 1]
```diff
@@ -149,8 +149,12 @@ class DiagnosticsWatcher {
         filePathsToIgnore: string[],
         ignoreInitialAdd: boolean
     ) {
-        watch(`${workspaceUri.fsPath}/**/*.{svelte,d.ts,ts,js,jsx,tsx,mjs,cjs,mts,cts}`, {
-            ignored: ['node_modules', 'vite.config.{js,ts}.timestamp-*']
+        watch(workspaceUri.fsPath, {
+            ignored: [
+                'node_modules',
+                'vite.config.{js,ts}.timestamp-*',
```
Comment (dummdidumm) [inline packages/svelte-check/src/index.ts:155]: Is this correct? AFAIK `node_modules` meant that everything with node_modules in the path was ignored, right? If not, we should change that.
The vite config string also looks suspiscuous, it's still using glob syntax, does this need to turn into a regex?

Comment (dummdidumm) [reply packages/svelte-check/src/index.ts:155]: Ok this has a couple of more bugs, will fix

**Comment by benmccann** [comment]
awesome. thanks for the cleanup!

**Pull request merged**